cfg/optimize.go

```diff
@@ -229,9 +231,11 @@ func getOptimizedValue(
 	currentValue any,
 	profileName string,
 	machineType string,
+	input OptimizationInput,
 	machineTypeToGroupMap map[string]string,
 ) OptimizationResult {
-	// Precedence: Profile -> Machine -> Default
+	// Precedence: Profile -> (Machine / Bucket) -> Default
```

Comment:
As discussed offline, let's not be ambiguous between machine-type and bucket-type .

```suggestion
	// Precedence: Profile > Machine-type > Bucket-Type > Default-Value
```